pycloudlib/ibm_softlayer/cloud.py

```diff
@@ -0,0 +1,492 @@
+from pycloudlib.ibm_softlayer.instance import IBMSoftlayerInstance
+from pycloudlib.instance import BaseInstance
+
+
+class IBMSoftlayer(BaseCloud):
+    """IBM Softlayer Class."""
+
+    _type = "ibmsoftlayer"
+
+    VALID_REGIONS = [
+        "osa",
+        "ams",
+        "che",
+        "tor",
+        "mon",
+        "syd",
+        "fra",
+        "tok",
+        "lon",
+        "sao",
+        "mad",
+        "wdc",
+        "par",
+        "dal",
+        "sjc",
+        "mil",
+        "sng",
+    ]
+
+    def __init__(
+        self,
+        tag: str,
+        timestamp_suffix: bool = True,
+        config_file: Optional[ConfigFile] = None,
+        *,
+        sl_username: Optional[str] = None,
+        sl_api_key: Optional[str] = None,
+    ):
+        """Initialize base cloud class.
+
+        Args:
+            tag: string used to name and tag resources with
+            timestamp_suffix: Append a timestamped suffix to the tag string.
+            config_file: path to pycloudlib configuration file
+            sl_username: IBM Softlayer username
+            sl_api_key: IBM Softlayer API key
+        """
+        super().__init__(
+            tag,
+            timestamp_suffix,
+            config_file,
+            required_values=[sl_username, sl_api_key],
+        )
+        self.created_keys: List[str] = []
+        self.created_security_groups: list = []
+
+        self._sl_username = sl_username or self.config.get("sl_username")
+        self._sl_api_key = sl_api_key or self.config.get("sl_api_key")
+
+        self._log.debug("logging into IBM")
+
+        self._client = SoftLayer.create_client_from_env(
+            username=self._sl_username,
+            api_key=self._sl_api_key,
+        )
+        self._virtual_server_manager = SoftLayer.VSManager(client=self._client)
+        self._image_manager = SoftLayer.ImageManager(client=self._client)
+        self._ssh_key_manager = SoftLayer.SshKeyManager(client=self._client)
+        self._network_manager = SoftLayer.NetworkManager(client=self._client)
+
+    def delete_image(self, image_id: str, **kwargs):
+        """Delete an image.
+
+        Args:
+            image_id: string, id of the image to delete
+            **kwargs: dictionary of other arguments to pass to delete_image
+        """
+        self._image_manager.delete_image(image_id)
+
+    def released_image(self, release, *, disk_size: str = "25G", **kwargs):
+        """ID (globalIdentifier) of the latest released image for a particular release.
+
+        Args:
+            release: The release to look for
+
+        Returns:
+            A single string with the latest released image ID for the
+            specified release.
+
+        """
+        public_images_gen = self._image_manager.list_public_images(
+            name=f"*{release}*"
+        )
+        public_images = list(public_images_gen)
+        if not public_images:
+            raise Exception(f"No public images found for {release}")
```

Comment:
In order to respect https://github.com/canonical/pycloudlib/issues/263, could we raise more concrete exceptions? 

```suggestion
            raise IBMSoftlayerException(f"No public images found for {release}")
```

Please, make sure this holds for the whole PR.

Reply:
thanks for pointing this out! i didnt realize how sloppy i was being with those.

Reply:
There are still a couple of `ValueError`s that probably would be better as an `IBMSoftlayerException` exception.